src/windows.rs

```diff
@@ -256,3 +264,22 @@ thread_local! {
 fn ensure_com_initialized() {
     CO_INITIALIZER.with(|_| {});
 }
+
+fn traverse_paths_recursively(
+    paths: impl IntoIterator<Item = PathBuf>,
+    collection: &mut Vec<PathBuf>,
+) -> Result<(), Error> {
+    for base_path in paths {
+        if base_path.is_file() {
+            collection.push(base_path);
+            continue;
+        }
+
+        for entry in fs::read_dir(&base_path).map_err(|err| Error::Unknown { description: err.to_string() })? {
+            let entry = entry.map_err(|err| Error::Unknown { description: err.to_string() })?;
+            traverse_paths_recursively(Some(entry.path()), collection)?;
+        }
+        collection.push(base_path);
+    }
```

Comment:
@Byron, @bicarlsen, I think this implementation is missing a check for `base_path.is_symlink()` and push that to the collection directly.

This reason this is necessary is because on Windows if `base_path` is a symlink to a directory `fs::read_dir` will iterate over the contents of the linked directory, leading to the removal of the contents of the directory and the symlink (not the original directory). This, I think, is incorrect.

Reply:
Nice catch. Maybe best to open a separate PR for this so we can also add a new test. Thoughts?

Reply:
That's a great catch indeed! Even though symlinks are less common on windows, traversing into symlinks is a critical flaw in the rare case of such a symlink being present.

Further, I have yanked v3.0.5, [added the `base_path.is_symlink()` check](https://github.com/Byron/trash-rs/pull/80) and created [a new release](https://github.com/Byron/trash-rs/releases/tag/v3.0.6).